### DISCUSSION TITLE
API fields explanation request

### DISCUSSION THREAD

**Discussion by kreolsky** [oobabooga/text-generation-webui#3926] [Q&A]
Hi, friends!
Can anyone explain below fields in API?

- 'max_tokens_second': 0, # What is it?
- 'mode': 'instruct',  # Valid options: 'chat', 'chat-instruct', 'instruct' How it works?
- 'character': 'Example', # Is it character template for 'mode': 'chat'? Looks like it does not work.
- 'your_name': 'You',
- 'name1': 'name of user', # What is difference with 'your_name'?
- 'name2': 'name of character', 
- 'name1_instruct': 'You', # Is it for instruct mode? How it works?
- 'name2_instruct': 'Assistant', # Optional
- 'context_instruct': 'context_instruct', # Optional
- 'turn_template': 'turn_template', # I have no idea about it :)
- 'regenerate': False, # I have no idea about it :)

**Comment by Kinglord** [comment]
I tried to ask this on reddit as well but didn't get any response and was downvoted as usual :D I was in particular looking at the chat stream API but you can see my best guesses for some of them here
https://www.reddit.com/r/Oobabooga/comments/16cpaus/ooba_api_prompt_constructiondocumentation/

In terms of your list

`'max_tokens_second': 0, # What is it?`
Answered below :D

`'mode': 'instruct', # Valid options: 'chat', 'chat-instruct', 'instruct' How it works?`
This sets the mode that ooba uses for the generation, same as the options in the webui

`'character': 'Example', # Is it character template for 'mode': 'chat'? Looks like it does not work.`
This will load a character you have saved on your server, this worked fine for me, but I don't know the actual construction of the character prompt/data it sends.

`'name1': 'name of user', # What is difference with 'your_name'?`
I wasn't sure here either, but I think it's an override for what's in the saved character, maybe?

`'name1_instruct': 'You', # Is it for instruct mode? How it works?`
AFAIK yes, I think it's the same as above.

`'turn_template': 'turn_template', # I have no idea about it :)`
This is for templates that like to have an idea of rounds or turns in a chat. I haven't used it yet though so I'm not sure the instruct format.

`'regenerate': False, # I have no idea about it :)`
Me either, I don't know if this is like a retry or something else.

**Reply by kreolsky** [reply]
'max_tokens_second' - It's limit for generation speed. "Maximum number of tokens/second"
